config/main.py

```diff
@@ -707,6 +710,51 @@ def reload():
     else:
         click.secho('Buffer definition template not found at {}'.format(buffer_template_file), fg='yellow')
 
+#
+# 'snmp-community' command
+#
+@config.command('snmp-community')
+@click.argument('community-name', metavar='<community_name>', required=True)
+def snmp_community(community_name):
+    """Set SNMP community string"""
+
+    """Serialization of a Python dict from a SNMP YAML File """
+    with open(SNMP_COMMUNITY_FILE, 'r') as stream:
+        try:
+            snmp_community_dict = yaml.safe_load(stream)
+        except yaml.YAMLError as error:
+            click.echo("SNMP config file not loaded correctly with error {}".format(error))
+            raise
+
+    if 'snmp_rocommunity' in snmp_community_dict.keys():
+        community_exist_name = snmp_community_dict['snmp_rocommunity']
+        del snmp_community_dict['snmp_rocommunity']
```

Comment:
>del [](start = 8, length = 3)

Prefer keep old data if possible. If both provided, snmp_rocommunities will take priority.

Reply:
Yes, I am keeping the old community string  i.e. `public`. I am just replacing the existing  key name `snmp_rocommunity` with a new key name `snmp_rocommunities` and adding the existing value i.e `public` to the new key name `snmp_rocommunities`.

However, if `snmp_rocommunities` is already present in `/etc/sonic/snmp.yml`, that means we have multiple community string values attached to the `snmp_rocommunities` key, so we will append the new community value to the key.

Hope this makes sense.

Reply:
I mean *not* `del snmp_community_dict['snmp_rocommunity']`, and keep `snmp_rocommunity` unchanged. It will be ignored since `snmp_rocommunities` is provided.

Reply:
If I get you right, this will be the workflow whenever we add a new SNMP community string i.e "**new_public**" to _/etc/sonic/snmp.yml_ file.


**For example**

before
```
snmp_rocommunity:
- public
```
After
```
snmp_rocommunity:
- public
snmp_rocommunities:
- new_public
```
Kindly verify if I understand correctly, I will modify my code accordingly.

Reply:
before
```
snmp_rocommunity:
- public
```
After
```
snmp_rocommunity:
- public
snmp_rocommunities:
- public, new_public
```